subprojects/packagefiles/libssh2/meson.build

```diff
@@ -0,0 +1,153 @@
+project('libssh2', 'c',
+  version : '1.10.0',
+  license : 'BSD-3',
+  meson_version : '>=0.58.0')
+
+cc = meson.get_compiler('c')
+pkgconfig = import('pkgconfig')
+
+libssh2_sources = [
+    'src/misc.c',
+    'src/scp.c',
+    'src/crypt.c',
+    'src/hostkey.c',
+    'src/version.c',
+    'src/wincng.c',
+    'src/blowfish.c',
+    'src/mac.c',
+    'src/os400qc3.c',
+    'src/transport.c',
+    'src/agent.c',
+    'src/pem.c',
+    'src/kex.c',
+    'src/sftp.c',
+    'src/knownhost.c',
+    'src/publickey.c',
+    'src/channel.c',
+    'src/global.c',
+    'src/packet.c',
+    'src/bcrypt_pbkdf.c',
+    'src/userauth.c',
+    'src/keepalive.c',
+    'src/comp.c',
+    'src/session.c'
+]
+
+libssh2_deps = []
+extra_flags = []
+
+if get_option('crypto_backend') == 'openssl'
+    libssh2_sources += ['src/openssl.c']
+    libssh2_deps += dependency('openssl', version : '>= 1.0.0')
+    extra_flags += ['-DLIBSSH2_OPENSSL']
+elif get_option('crypto_backend') == 'mbedtls'
+    libssh2_sources += ['src/mbedtls.c']
+
+    if dependency('mbedtls', required: false).found()
+        libssh2_deps += dependency('mbedtls')
+    else
+        libmbedtls = cc.find_library('mbedtls', has_headers: ['mbedtls/version.h'])
+        libmbedx509 = cc.find_library('mbedx509')
+        libmbedcrypto = cc.find_library('mbedcrypto')
+        libssh2_deps += [libmbedtls, libmbedx509, libmbedcrypto]
+    endif
```

Comment:
But mbedtls can't ever exist, can it?

And even if it did, you should save the dependency() once, instead of looking it up a second time after checking if it got found the first time.

Reply:
Hm, `compiler.find_library()` has `required: true` by default so no `mbedtls` in the system will cause the following error:
> Run-time dependency mbedtls found: NO (tried pkgconfig and cmake)

Reply:
I'm not sure why that error is a problem? Don't we want that? If the mbedtls backend is selected, then its absence *should* be an error, no?

Also, it's still not clear to me how `dependency('mbedtls')` could exist since the upstream mbedtls project doesn't install any dependency exports.

Reply:
Oh, I get it now, this dependency('mbedtls') was added just in case the upstream will add dependency exports in future.

Reply:
@eli-schwartz so do you think it's a bad idea and I should remove it?

Reply:
Well, it's a wasted lookup, produces confusing logs, and we can always add it if and when mbedtls upstream adds support.

Reply:
Ok, agree, removed it.